### PR TITLE
Feature/5947 tests contact journal extension/7406 add coronatest to contact journal

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
add test result to the diary if the result is positive or negative

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7406

### CODE REVIEW

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 1]
```diff
@@ -1222,6 +1251,161 @@ class CoronaTestServiceTests: XCTestCase {
 		XCTAssertEqual(mockNotificationCenter.notificationRequests.count, 0)
 	}
 
+	func test_When_UpdateTestResultSuccessWithPositive_Then_ContactJounalHasAnEntry() throws {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1254]:
```suggestion
	func test_When_UpdateTestResultSuccessWithPositive_Then_ContactJournalHasAnEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 2]
```diff
@@ -1222,6 +1251,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 2)
+		XCTAssertTrue(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertTrue(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithPending_Then_ContactJounalHasAnEntry() throws {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1285]:
```suggestion
	func test_When_UpdateTestResultSuccessWithPending_Then_ContactJournalHasAnEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 3]
```diff
@@ -1222,6 +1251,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 0)
+		XCTAssertFalse(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertFalse(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithExpired_Then_ContactJounalHasAnEntry() throws {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1316]:
```suggestion
	func test_When_UpdateTestResultSuccessWithExpired_Then_ContactJournalHasAnEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 4]
```diff
@@ -1222,6 +1251,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 0)
+		XCTAssertFalse(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertFalse(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithInvalid_Then_ContactJounalHasAnEntry() throws {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1347]:
```suggestion
	func test_When_UpdateTestResultSuccessWithInvalid_Then_ContactJournalHasAnEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 5]
```diff
@@ -1222,6 +1251,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 0)
+		XCTAssertFalse(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertFalse(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithNegativ_Then_ContactJounalHasAnEntry() throws {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1378]:
```suggestion
	func test_When_UpdateTestResultSuccessWithNegativ_Then_ContactJournalHasAnEntry() throws {
```

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 6]
```diff
@@ -1215,6 +1244,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 2)
+		XCTAssertTrue(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertTrue(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithPending_Then_ContactJournalHasAnEntry() throws {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1278]:
```suggestion
	func test_When_UpdateTestResultSuccessWithPending_Then_ContactJournalHasNoEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 7]
```diff
@@ -1215,6 +1244,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 0)
+		XCTAssertFalse(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertFalse(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithExpired_Then_ContactJournalHasAnEntry() throws {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1309]:
```suggestion
	func test_When_UpdateTestResultSuccessWithExpired_Then_ContactJournalHasNoEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 8]
```diff
@@ -1215,6 +1244,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 0)
+		XCTAssertFalse(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertFalse(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithInvalid_Then_ContactJournalHasAnEntry() throws {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1340]:
```suggestion
	func test_When_UpdateTestResultSuccessWithInvalid_Then_ContactJournalHasNoEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift [hunk 9]
```diff
@@ -1215,6 +1244,161 @@ class CoronaTestServiceTests: XCTestCase {
...
+			completionExpectation.fulfill()
+		}
+		waitForExpectations(timeout: .short)
+
+		XCTAssertEqual(diaryStore.coronaTests.count, 0)
+		XCTAssertFalse(try XCTUnwrap(testService.antigenTest?.journalEntryCreated))
+		XCTAssertFalse(try XCTUnwrap(testService.pcrTest?.journalEntryCreated))
+	}
+
+	func test_When_UpdateTestResultSuccessWithNegativ_Then_ContactJournalHasAnEntry() throws {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift:1371]:
```suggestion
	func test_When_UpdateTestResultSuccessWithNegative_Then_ContactJournalHasAnEntry() throws {
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/MockDiaryStore.swift [hunk 10]
```diff
@@ -261,7 +261,10 @@ class MockDiaryStore: DiaryStoringProviding {
 				DiaryDay(
 					dateString: dateString,
 					entries: contactPersonEntries + locationEntries,
-					tests: []
+					tests: coronaTests
+						.filter { diaryDayTest in
+					  diaryDayTest.date == dateString
+				  }
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/MockDiaryStore.swift:267]: Just GitHub or is the indentation off?

**Review by nickguendling — APPROVED** [review]
